### PR TITLE
Feature/logout and fields feedback login

### PR DESCRIPTION
**Pull request by brunomichetti**
- Logout feature
- Separate userProfile in different component
- Delete alert for successful login
- Add field validation to email and password in login
- Improvement of name colors using [name that color](http://chir.ag/projects/name-that-color/#6195ED)

- Feedback for wrong email or password:
Screen Shot 2020-06-18 at 16 52 15

- Feedback for missing fields:
Screen Shot 2020-06-18 at 16 51 58

### CODE REVIEW

File: src/actions/user.actions.js [hunk 1]
```diff
@@ -11,18 +11,28 @@ const login = (email, password) => {
...
 
+const logout = () => {
+    return async dispatch => {
+        dispatch(request());
+        await userService.logout();
+        dispatch(logoutResult);
+        history.push("/");
+    };
+    function request() { return { type: userConstants.LOGOUT_REQUEST } }
+    function logoutResult() { return { type: userConstants.LOGOUT } }
```
Comment (tamarasuarezrod) [inline src/actions/user.actions.js]: is this working? I think the code after a return is unreachable

File: src/components/users/LoginForm.js [hunk 2]
```diff
@@ -16,8 +16,16 @@ const LoginForm = () => {
     const { email, password } = inputs;
     // cleanAlert is to remove error alert on change. This will be erased
     const [cleanAlert, setCleanAlert] = useState(false);
+    const [missingEmail, setMissingEmail] = useState(false);
+    const [missingPassword, setMissingPassword] = useState(false);
     // function to update the inputs when the user modifies the input
     function handleChange(e) {
```
Comment (tamarasuarezrod) [inline src/components/users/LoginForm.js]:
```suggestion
    const handleChange = ({ target }) => {
```

File: src/components/users/LoginForm.js [hunk 3]
```diff
@@ -28,25 +36,33 @@ const LoginForm = () => {
     const dispatch = useDispatch();
     // handle submit of the form
     function handleSubmit(e) {
```
Comment (tamarasuarezrod) [inline src/components/users/LoginForm.js]: same here

File: src/pages/HomePage.js [hunk 4]
```diff
@@ -1,42 +1,12 @@
...
-                    <a href="/home"><FormattedMessage id="homepage.edit.text"/></a> &nbsp; / &nbsp;
-                    <a href="/home"><FormattedMessage id="homepage.logout.text"/></a>
-                </div>
-                <hr className="left-menu__hr"></hr>
-                <p className="left-menu__matches"><FormattedMessage id="homepage.nomatches.text"/></p>
-                <div className="left-menu__smilies">
-                    <img src={smilies} alt="smilies" className="smilies-img"></img>
-                </div>     
-            </div>       
+            <UserProfile />
```
Comment (tamarasuarezrod) [inline src/pages/HomePage.js:9]: 👏

File: src/reducers/auth.reducer.js [hunk 5]
```diff
@@ -4,20 +4,26 @@ let user = JSON.parse(localStorage.getItem("user"));
...
+        case userConstants.LOGIN_REQUEST:
+            return {
+                loggingIn: true,
+                user: user
+            };
+        case userConstants.LOGIN_SUCCESS:
+            return {
+                loggedIn: true,
+                user: user
+            };
```
Comment (tamarasuarezrod) [inline src/reducers/auth.reducer.js:17]:
```suggestion
        case userConstants.LOGIN_REQUEST:
        case userConstants.LOGIN_SUCCESS:
            return {
                loggedIn: true,
                user: user
            };
```
A shortcut to repeat the same code in more than one case

File: src/reducers/auth.reducer.js [hunk 6]
```diff
@@ -4,20 +4,26 @@ let user = JSON.parse(localStorage.getItem("user"));
...
+        case userConstants.LOGIN_FAILURE:
+            return {};
```
Comment (tamarasuarezrod) [inline src/reducers/auth.reducer.js:19]: do you need this case? Does nothing

File: src/reducers/auth.reducer.js [hunk 7]
```diff
@@ -4,20 +4,26 @@ let user = JSON.parse(localStorage.getItem("user"));
...
+        case userConstants.LOGOUT_REQUEST:
+            return {
+                loggingOut: true
+            };
+        case userConstants.LOGOUT:
+            return {};
```
Comment (tamarasuarezrod) [inline src/reducers/auth.reducer.js:24]: You can do the same with this and `LOGIN_FAILURE`

File: src/components/users/UserProfile.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+import { userActions } from "../../actions/user.actions";
+
+const UserProfile = () => {
+    const dispatch = useDispatch();
+    const { user: { email }} = JSON.parse(localStorage.getItem("user"));
+    const loggingOut = useSelector(state => state.authentication.loggingOut);
+    function handleLogout() {
+        dispatch(userActions.logout())
+    }
+    return (
```
Comment (tamarasuarezrod) [inline src/components/users/UserProfile.js:21]:
```suggestion
    const handleLogout = () => {
        dispatch(userActions.logout())
    }
    
    return (
```

File: src/components/users/user-profile.scss [hunk 9]
```diff
@@ -1,51 +1,75 @@
...
+
+.logout__btn-text:hover {
+    // Underline and turn the button blue on hover
+    text-decoration: underline;
+    color: $dodger-blue;
+}
+
+.logout__btn-text:focus{
+    // Eliminate border when the button is clicked
+    border-color:transparent!important;
```
Comment (manema) [inline src/components/users/user-profile.scss]: try to avoid use !important as long as possible, it's considered a bad practice since could lead in side effects. There will be some special cases when you won't have another way to go, but in general you will be able to avoid them just restructuring your css.

File: src/helpers/private.route.js [hunk 10]
```diff
@@ -2,17 +2,17 @@ import React from "react";
 import { useSelector } from "react-redux";
 import { Redirect, Route } from "react-router-dom";
 
-const PrivateRoute = ({ component: Component, ...rest }) => {
+const PrivateRoute = ({ comp: Component, ...rest }) => {
```
Comment (manema) [inline src/helpers/private.route.js:5]: Why this change? just being curious

File: src/components/users/LoginForm.js [hunk 11]
```diff
@@ -16,8 +16,16 @@ const LoginForm = () => {
     const { email, password } = inputs;
     // cleanAlert is to remove error alert on change. This will be erased
     const [cleanAlert, setCleanAlert] = useState(false);
+    const [missingEmail, setMissingEmail] = useState(false);
+    const [missingPassword, setMissingPassword] = useState(false);
     // function to update the inputs when the user modifies the input
     function handleChange(e) {
+        const field = e.target.name;
+        if (field === "email") {
```
Comment (manema) [inline src/components/users/LoginForm.js]: I guess you can omit this piece of code and the one from line 39 to 44. Then, in lines 57 and 60 evaluate inputs' values themselves. If a field is "" it means it's empty, it is not otherwise.

**Review by manema — APPROVED** [review]

File: src/actions/user.actions.js (hunk 1 above)
Comment (brunomichetti) [reply src/actions/user.actions.js]: Yes it's working. Maybe move this above?

File: src/components/users/user-profile.scss (hunk 9 above)
Comment (brunomichetti) [reply src/components/users/user-profile.scss]: And what is a good practice to remove the border of a button?

File: src/helpers/private.route.js (hunk 10 above)
Comment (brunomichetti) [reply src/helpers/private.route.js:5]: Because a warning was thrown with Private route, and I tried something I saw in the bible (stackoverflow). And it worked haha

File: src/components/users/user-profile.scss (hunk 9 above)
Comment (manema) [reply src/components/users/user-profile.scss]: I guess that playing around with border-color, outline and box-shadow css properties you will be able to achieve the expected behavior, I'm discouraging the use of **!important** . doesn't it work if you remove it? Maybe you need it to override the browser default style under _focus_ state, in this case it's ok leave it, but you should be able to set background and padding in lines 52 and 55 without a !important tag.

File: src/components/users/LoginForm.js [hunk 12]
```diff
@@ -8,6 +8,7 @@ import "../../style/App.scss";
 import "./login-form.scss";
 
 const LoginForm = () => {
+    debugger;
```
Comment (tamarasuarezrod) [inline src/components/users/LoginForm.js]: remove this

File: src/components/users/LoginForm.js [hunk 13]
```diff
@@ -16,37 +17,42 @@ const LoginForm = () => {
...
+            {isSubmitted && !email && <div className="login-form__alert"><FormattedMessage id="loginform.missing.email.text"/></div>}
             <p className="login-form__text">PASSWORD</p>
             <input className="login-form__input" type="password" name="password" value={password} onChange={handleChange} autoComplete="on"/>
+            {isSubmitted && !password && <div className="login-form__alert"><FormattedMessage id="loginform.missing.pass.text"/></div>}
             <div>
                 <button type="submit" className="login-form__btn-text"><FormattedMessage id="loginform.signin.text"/></button>
             </div>
             {loggingIn && <Loader type="ThreeDots" color="#2FBCF7" height={80} width={50}/>}
-            {!loggingIn && !(cleanAlert) && alert && alert.message && <div>{alert.message}</div>}
+            {isSubmitted && !loggingIn && !cleanAlert && alert && <div className="login-form__alert"> {alert.message} </div>}
```
Comment (tamarasuarezrod) [inline src/components/users/LoginForm.js]: maybe you should move `isSubmitted && !loggingIn && !cleanAlert && alert` to a constant, `showLoginAlert`?

File: src/reducers/alert.reducers.js [hunk 14]
```diff
@@ -2,16 +2,12 @@ import { alertConstants } from "../constants/alert.constants";
...
-      return {
-        type: "alert-success",
-        message: action.message
-      };
     case alertConstants.ERROR:
       return {
         type: "alert-danger",
         message: action.message
       };
+    case alertConstants.SUCCESS:
```
Comment (tamarasuarezrod) [inline src/reducers/alert.reducers.js]: is this needed? given that is the same as the default

**Review by tamarasuarezrod — APPROVED** [review]
